lib/xayn_card_view/card_view.dart

```diff
@@ -292,41 +312,55 @@ class _CardViewState extends CardViewAnimatedState with CardViewListenersMixin {
   List<Widget> _buildVisibleCards({
     required double itemSpacing,
     required BorderRadius clipBorderRadius,
+    required BuildContext context,
     double? width,
     double? height,
-  }) =>
-      [
-        if (index > 0)
-          _buildCard(
-            index: index - 1,
-            builder: widget.secondaryItemBuilder,
-            itemSpacing: itemSpacing,
-            borderBuilder: widget.borderBuilder,
-            clipBorderRadius: clipBorderRadius,
-            width: width,
-            height: height,
-          ),
-        if (widget.itemCount > 0)
-          _buildCard(
-            index: index,
-            builder: widget.itemBuilder,
-            itemSpacing: itemSpacing,
-            borderBuilder: widget.borderBuilder,
-            clipBorderRadius: clipBorderRadius,
-            width: width,
-            height: height,
-          ),
-        if (index < widget.itemCount - 1)
-          _buildCard(
-            index: index + 1,
-            builder: widget.secondaryItemBuilder,
-            itemSpacing: itemSpacing,
-            borderBuilder: widget.borderBuilder,
-            clipBorderRadius: clipBorderRadius,
-            width: width,
-            height: height,
-          ),
-      ];
+  }) {
+    // resolvedIndex is the actual data index, without a maybe-included finalItemBuilder card
+    final resolvedIndex = index > 0 ? index.clamp(0, widget.itemCount - 1) : 0;
```

Comment:
I think having `widget.itemCount - 1` in a variable `lastIndex` would make the code more readable. But not necessary ofc.